### ISSUE TITLE
Android: Crash marking folder read from long press

### ISSUE THREAD

**Issue by manderson23** [samuelclay/NewsBlur#911]
```
java.lang.ArrayIndexOutOfBoundsException: length=12; index=-1
  at java.util.ArrayList.get(ArrayList.java:310)
  at com.newsblur.database.FolderListAdapter.getChild(FolderListAdapter.java:336)
  at com.newsblur.fragment.FolderListFragment.onContextItemSelected(FolderListFragment.java:273)
  at android.app.Fragment.performContextItemSelected(Fragment.java:2174)
  at android.app.FragmentManagerImpl.dispatchContextItemSelected(FragmentManager.java:2002)
  at android.app.Activity.onMenuItemSelected(Activity.java:2908)
```
@dosiecki from a quick look at the code are folders no longer handled correctly after the recent refactoring for saved stories?

**Comment by dosiecki** [comment]
The new code for the feed list hasn't even been beta tested yet, so there could certainly be some bugs.  Good catch; I will take a look tomorrow.

**Comment by dosiecki** [comment]
fixed in #912
